eslint/babel-eslint-plugin-internal/src/rules/dry-error-messages.js

```diff
@@ -0,0 +1,144 @@
+import path from "path";
+
+const REL_PATH_REGEX = /^\.{1,2}/;
+
+function isRelativePath(filePath) {
+  return REL_PATH_REGEX.test(filePath);
+}
+
+function normalizeSrcExt(source) {
+  return !path.extname(source) ? `${source}.js` : source;
+}
+
+function isErrorModule(targetModulePath, currentFilePath, src) {
+  if (isRelativePath(src)) {
+    return (
+      path.normalize(targetModulePath) ===
+      path.normalize(
+        normalizeSrcExt(path.resolve(path.dirname(currentFilePath), src)),
+      )
+    );
+  }
+
+  return targetModulePath === src;
+}
+
+function findIdNode(node) {
+  if (node.type === "Identifier") {
+    return node;
+  }
+
+  if (node.type === "MemberExpression" && node.object.type === "Identifier") {
+    return node.object;
+  }
+
+  return null;
+}
+
+function findReference(scope, node) {
+  let currScope = scope;
+
+  while (currScope) {
+    const ref = currScope.set.get(node.name);
+
+    if (ref) {
+      return ref;
+    }
+
+    currScope = currScope.upper;
+  }
+
+  return null;
+}
+
+function referencesImportedBinding(node, scope, bindings) {
+  const ref = findReference(scope, node);
+
+  if (ref) {
+    const topLevelDef = ref.defs[0];
+
+    if (topLevelDef.type === "ImportBinding") {
+      const defNode = topLevelDef.node;
+
+      for (const spec of bindings) {
+        if (
+          spec.loc.start === defNode.loc.start &&
+          spec.loc.end === defNode.loc.end
+        ) {
+          return true;
+        }
+      }
+    }
+  }
+
+  return false;
+}
+
+export default {
+  meta: {
+    type: "suggestion",
```

Comment:
Does `suggestion` make CI fail? (I'd like it to do so)

Reply:
In this case, `"suggestion"` is referring to what type of rule this is (i.e. it enforces a code style choice rather than a possible code error). It can still be configured as `"off"`, `"warning"`, or `"error"` in our config :)